questionpy_sdk/webserver/state_storage.py

```diff
@@ -0,0 +1,61 @@
+import json
+from pathlib import Path
+from typing import Optional, Union, List
+
+from questionpy_common.elements import StaticTextElement, SelectElement, GroupElement, OptionsFormDefinition, \
+    FormElement, RepetitionElement
+
+
+class QuestionStateStorage:
+
+    def __init__(self) -> None:
+        # Mapping of package path to question state path
+        self.paths: dict[Path, Path] = {}
+        self.storage_path: Path = Path(__file__).parent / 'question_state_storage'
+
+    def insert(self, key: Path, question_state: dict) -> None:
+        path = self.storage_path / key.with_suffix('.json')
+        self.paths[key] = path
+        with path.open('w') as file:
+            json.dump(question_state, file)
+
+    def get(self, key: Path) -> Optional[dict]:
+        if key in self.paths:
+            path = self.paths.get(key)
+            if path and path.exists():
+                return json.loads(path.read_text())
+        path = self.storage_path / key.with_suffix('.json')
+        if not path or not path.exists():
+            return None
+        self.paths[key] = path
+        return json.loads(path.read_text())
+
+    def parse_form_data(self, form_definition: OptionsFormDefinition, form_data: dict) -> dict:
+        options = self._parse_section(form_definition.general, form_data)
+        return options
```

Comment:
```suggestion
        return self._parse_section(form_definition.general, form_data)
```

Da `_parse_section()` nur hier verwendet wird und nur das Ergebnis des Methodenaufrufs zurückgegeben wird, könnte man eigentlich direkt die Logik in `parse_form_data()` platzieren.